source/manual/common-aws-tasks-for-2nd-line-support.html.md

```diff
@@ -0,0 +1,235 @@
+3. In the AWS UI, cycle the running instances gradually, to avoid downtime.
+
+  Cycling an instance might involve removing the smaller instances one by one,
+  waiting for connections to the instance to drain, and then killing the
+  instance. This will prompt AWS to bring up the new bigger instances
+  automatically.
+
+  Be careful here, provisioning a machine is slow, flaky and
+  requires manual intervention (e.g. sometimes we need to manually run puppet
+  on a new instance, must deploy apps to new instances from Jenkins).
+  Check that the new instances are healthy before removing the healthy old
+  instances.
+
+## How to scale horizontally
+
+You can increase the number of EC2 instances running in an ASG via a config
+change in govuk-aws and govuk-aws-data.
+
+In an emergency, you can also manually bump up the number of instances in
+the ASG via the AWS console, but you're encouraged to use Terraform to do this,
+since you'll need to make the change in Terraform anyway.
+
+Here's an example set of PRs:
+
+* https://github.com/alphagov/govuk-aws-data/pull/805
+* https://github.com/alphagov/govuk-aws/pull/1385
+
+You will need to apply Terraform for this to have an effect.
+
+## How to resize a persistent disk
+
+If you're not sure how to do this, ask an Site Reliability Engineer to give you a walk through.
+
+See the docs: https://docs.publishing.service.gov.uk/manual/manually-resize-ebs.html.
+
+## How to replace an instance with a persistent disk
+
+_If you're not sure how to do this, ask an Site Reliability Engineer to give
+you a walk through._
+
+Assuming we want to destroy and recreate a VM, but have the new VM attach to the
+old persistent disk
+
+1. Find the VM and note down the details of the persistent volume you want to keep
+1. Shutdown the VM
+1. Detach the EBS volume from the VM.
+1. Delete the VM
+1. Run terraform apply which will now recreate the VM
+1. Manually re-attach the EBS volume to the new VM
+1. Reboot the machine to make sure the persistent disk reattaches between reboots
+1. Initiate a puppet run to make sure that everything works
+
+## Learn
+
+### How do we do DNS?
+
+See the documentation on [how GOV.UK does DNS][].
+
+GOV.UK manages the DNS for other domain names used by third parties, for example
+service.gov.uk. At the moment this will remain with the SREs until further
+decisions have been made.
+
+[how GOV.UK does DNS]: https://docs.publishing.service.gov.uk/manual/dns.html
+
+## How are instances provisioned?
+
+We typically use Terraform config in [govuk-aws][] to provision infrastructure,
+such as EC2 instances.
+
+There are a few exceptions to this, such as ad-hoc instances started from
+Concourse via the AWS CLI - these are mainly for data science projects.
+
+We use [userdata scripts][] to run commands on our instances at launch. These
+scripts install various core bits of software needed by a particular instance
+and then typically use [govuk-puppet][] to provision our instances.
```

Comment:
we need to add that part of the start-up, machines will contact Jenkins with their FDQN and puppet class so that GOV.UK apps are provisioned by Jenkins.

Refs:
[1] Jenkins job AWS Integration: https://deploy.integration.publishing.service.gov.uk/job/Deploy_Node_Apps/
[2] Jenkins trigger in Puppet: https://github.com/alphagov/govuk-puppet/blob/8cca9aad2b68d6cb396a135f47524fafeca1c947/modules/govuk/manifests/deploy/sync.pp